elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -254,19 +253,35 @@ private SQLQuery toPageTotalSQL(SQLQuery sql) {
      *
      * @param path The predicate path to convert
      * @param dictionary dictionary to expand joinTo path
+     * @param expressions outer layer sql expression to be applied on this field
      * @return A SQL fragment that references a database column
      */
-    public static String generateColumnReference(Path path, EntityDictionary dictionary) {
+    public static String generateColumnReference(Path path, EntityDictionary dictionary, List<String> expressions) {
```

Comment:
Instead of List<String>, I think we want more of a symbol table concept here with concrete names that map to the model.